src/Microsoft.Health.Dicom.Core/Features/Store/StoreValidatorResultBuilder.cs

```diff
@@ -0,0 +1,89 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.ObjectModel;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+
+namespace Microsoft.Health.Dicom.Core.Features.Store;
+
+internal sealed class StoreValidatorResultBuilder
+{
+    public StoreValidatorResultBuilder()
+    {
+        ErrorMessages = new Collection<string>();
```

Comment:
Do not use `Collection<T>`. This is a base class used by other classes typically in the `System.Collections.ObjectModel` namespace. Please use a more typical collection like `List<T>`

Reply:
Tried using List<T>. But I get code analysis error:

CA1002	Change 'List<string>' in 'StoreValidationResultBuilder._errorMessages' to use 'Collection<T>', 'ReadOnlyCollection<T>' or 'KeyedCollection<K,V>'

Reply:
Are you sure about this? I could imagine you might get this error when they were properties, but not as private fields. Can you double-check? Otherwise, we can change our `.globalconfig`. We have the power

Reply:
resolved